models/roberta_encoder.py

```diff
@@ -12,9 +12,15 @@ def __init__(self):
         # this model is in eval per default, we want to fine-tune it but only the top layers
         self.model.train()
 
-        # only freezing the encoder parameters / weights of the head layers
-        for param in self.model.base_model.parameters():
-            param.requires_grad = False
+        # freeze half if we are doing hyper parameter search
```

Comment:
Shouldn't you always freeze half? When doing transfer learning we also want to still train the other half of the model right?

Reply:
During hyperparameter search we freeze half of the layers so it's faster. When doing the final transfer learning we finetune al the layers.

Reply:
Oh really? I thought Phillip recommended only training half of the layers

Reply:
As far as I understood his comment was for the hyperparameter search, and then finetune the whole model once the best configuration as been found (I could be wrong)

Reply:
Yes I now reread the assignment and you're correct